### PR TITLE
build: Drop OSGi support

### PR DESCRIPTION
**Pull request by johanandren**
References #28304 

I have also gone through OSGi related PRs and tried to see if there were any non-obvious OSGi changes to revert but couldn't really find anything.

### CODE REVIEW

File: akka-docs/src/main/paradox/additional/osgi.md [hunk 1]
```diff
@@ -1,116 +1,6 @@
 # Akka in OSGi
```
Comment (patriknw) [inline akka-docs/src/main/paradox/additional/osgi.md:1]: start a `akka-docs/src/main/paradox/project/migration-guide-2.7.x-2.8.x.md` page and mention the removal there.

I'm not sure we need to keep this osgi.md page at all (possibly redirect to migration guide)?

**Review by patriknw — COMMENTED** [review]
looking good, nice

File: akka-docs/src/main/paradox/additional/osgi.md (hunk 1 above)
Comment (johanandren) [reply akka-docs/src/main/paradox/additional/osgi.md:1]: A redirect is better 👍

File: project/Jdk9.scala [hunk 2]
```diff
@@ -41,6 +41,7 @@ object Jdk9 extends AutoPlugin {
     externalDependencyClasspath := (Test / externalDependencyClasspath).value)
 
   val compileSettings = Seq(
+    // FIXME see if this works now that we dropped OSGi
```
[the remark this answers is not in the document]
Comment (johanandren) [reply project/Jdk9.scala]: Verified with a +publishLocal that the jars still contain the JDK9+ classes with this change 👍

File: project/Jdk9.scala [hunk 3]
```diff
@@ -41,12 +41,8 @@ object Jdk9 extends AutoPlugin {
...
 
   val compileSettings = Seq(
-    // FIXME see if this works now that we dropped OSGi
-    // It might have been more 'neat' to add the jdk9 products to the jar via packageBin/mappings, but that doesn't work with the OSGi plugin,
-    // so we add them to the fullClasspath instead.
-    //    Compile / packageBin / mappings
-    //      ++= (CompileJdk9 / products).value.flatMap(Path.allSubpaths),
-    Compile / fullClasspath ++= (CompileJdk9 / exportedProducts).value)
+    // add the jdk9 products to the jar
+    Compile / packageBin / mappings ++= (CompileJdk9 / products).value.flatMap(Path.allSubpaths))
```
Comment (patriknw) [inline project/Jdk9.scala:45]: nice

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**